### PR TITLE
Adding slugify document type id

### PR DESCRIPTION
**Pull request by detautama**
This closes https://github.com/oddbit/tanam/issues/55
Slugify

image

### CODE REVIEW

File: angular/projects/admin/src/app/services/document-type.service.ts [hunk 1]
```diff
@@ -31,11 +31,11 @@ export class DocumentTypeService {
       .valueChanges();
   }
 
-  async create(id: string) {
-    const docRef = this.siteCollection.collection('document-types').doc(id);
+  async create(title: string, slug: string) {
```
Comment (DennisAlund) [inline angular/projects/admin/src/app/services/document-type.service.ts]: Try to not expand the object's field attributes into a list of parameters. Service APIs should as much as possible take either 

 1. The "object type" as a parameter
 2. Just a string `id`

E.g. either

```typescript
create('some-id')
```

or

```typescript
const documentType = {...} as DocumentType;
create(documentType)
```
If you need a function that does something in particular, then you make a function called `createWithTitle` that only takes the title and slugification is inside of the service.

Comment (DennisAlund) [reply angular/projects/admin/src/app/services/document-type.service.ts]: So change the name of `create` to `createWithTitle` and the only parameter to the function is the **non slugified** title. Move `_slugify` inside the service.

File: angular/projects/admin/src/app/services/document-type.service.ts [hunk 2]
```diff
@@ -31,11 +31,11 @@ export class DocumentTypeService {
...
+    const docRef = this.siteCollection.collection('document-types').doc(slug);
     return docRef.set({
-      id: id,
-      title: null,
+      id: slug,
+      title: title,
       description: '',
       slug: null,
```
Comment (DennisAlund) [inline angular/projects/admin/src/app/services/document-type.service.ts]: The slug should be saved to `slug` field also.

Comment (detautama) [reply angular/projects/admin/src/app/services/document-type.service.ts]: Resolve on this commit https://github.com/oddbit/tanam/pull/59/commits/8fca9d95aa49ab745896b6bf3d23e3cd48589233

File: angular/projects/admin/src/app/services/document-type.service.ts (hunk 1 above)
Comment (detautama) [reply angular/projects/admin/src/app/services/document-type.service.ts]: Resolve on this commit https://github.com/oddbit/tanam/pull/59/commits/8fca9d95aa49ab745896b6bf3d23e3cd48589233

File: angular/projects/admin/src/app/services/document-type.service.ts [hunk 3]
```diff
@@ -31,13 +33,15 @@ export class DocumentTypeService {
       .valueChanges();
   }
 
-  async create(title: string, slug: string) {
+  async createWithTitle(title: string) {
+    const slug = this._slugify(title);
     const docRef = this.siteCollection.collection('document-types').doc(slug);
+    this.router.navigateByUrl(`/_/admin/type/${slug}/edit`);
```
[the remark this answers is not in the document]
Comment (detautama) [reply angular/projects/admin/src/app/services/document-type.service.ts]: Resolve on this commit https://github.com/oddbit/tanam/pull/59/commits/2458f2bb8660f89f720c3144d8f77b46c52bac95

**Pull request merged**